### PR TITLE
Sync owners with k/release

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Part of https://github.com/kubernetes/sig-release/issues/2111
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm
/hold

there is one user that looks like is not part of this org, should be added before?

File: OWNERS [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
-  - jeremyrickard
-  - justaugustus
-  - puerco
-  - saschagrunert
+  - sig-release-leads
+  - release-engineering-approvers
 
 reviewers:
-  - leonardpahlke
   - encodeflush
```
Comment (leonardpahlke) [inline OWNERS:8]:
```suggestion
  - hosseinsalahi
```

**Review by leonardpahlke — COMMENTED** [review]
Hossein changed his username

File: OWNERS (hunk 1 above)
Comment (saschagrunert) [reply OWNERS]: Thanks! Fixed as suggested.

**Comment by saschagrunert** [comment]
> hosseinsalahi
> * User is not a member of the org. Satisfy at least one of these conditions to make the user trusted.

We need to change this here: https://github.com/kubernetes/org/blob/c16ba76c44efb74562531cb0e8c83879c3744842/config/kubernetes-sigs/org.yaml#L249

**Comment by saschagrunert** [comment]
Looks like the bot caches the org repo somewhere... 
/verify-owners

**Comment by saschagrunert** [comment]
@hosseinsalahi did you ever get an invite to the k-sigs org?

**Pull request merged**